notebooks/Run generative finetuning.ipynb

```diff
@@ -0,0 +1,373 @@
+    "algorithm_map = {\n",
+    "    \"us-west-2\": \"arn:aws:sagemaker:us-west-2:594846645681:algorithm/classification-finetuning-91717d633bb2357ba721ffc4ba2fe75c\",\n",
+    "    \"us-east-1\": \"arn:aws:sagemaker:us-east-1:865070037744:algorithm/classification-finetuning-91717d633bb2357ba721ffc4ba2fe75c\",\n",
+    "    \"eu-west-1\": \"arn:aws:sagemaker:eu-west-1:985815980388:algorithm/classification-finetuning-91717d633bb2357ba721ffc4ba2fe75c\",\n",
+    "}\n",
+    "if region not in algorithm_map.keys():\n",
+    "    raise Exception(f\"Current boto3 session region {region} is not supported.\")\n",
+    "\n",
+    "arn = algorithm_map[region]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### A. Upload training data\n",
+    "\n",
+    "Choose a directory on S3 to store the training data and add the link below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3_data_dir = \"s3://...\"  # Do not add a trailing slash otherwise the upload will not work"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Upload sample training data to S3:\n",
+    "\n",
+    "### Note:\n",
+    "\n",
+    "You'll need your data in a .csv, .jsonl or .txt file\n",
+    "\n",
+    "We strongly recommend using a .csv or .jsonl and having prompt-completion pairs as your examples for best results.\n",
+    "\n",
+    "If your data cannot be structured in prompt-completion pairs, you may leave your prompt fields blank\n",
+    "\n",
+    "\n",
+    "### Example:\n",
+    "\n",
+    "JSONL:  `{\"prompt\": \"This is the first prompt\", \"completion\": \"This is the first completion\"}`\n",
+    "\n",
+    "CSV:  `\"This is the first prompt\" , \"This is the first completion\"`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sess = sage.Session()\n",
+    "train_dataset = S3Uploader.upload(\"../examples/sample_sst5_finetuning_data.jsonl\", s3_data_dir, sagemaker_session=sess)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### B. Finetune model on uploaded data\n",
+    "\n",
+    "Specify a directory on S3 where finetuned models should be stored:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3_models_dir = \"s3://...\"  # Do not add a trailing slash otherwise it will not work"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create Cohere client:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "co = Client(region_name=region)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define parameters\n",
```

Comment:
Size and strategy will not be parameters in the listing. This was only for testing. Lets remove these from here and the hyperparameters

Reply:
ah, my bad. so no passing hyperparameters?

Reply:
Everything except size and strategy